src/azure-cli/azure/cli/command_modules/keyvault/_client_factory.py

```diff
@@ -134,7 +134,8 @@ def keyvault_data_plane_factory(cli_ctx, _):
     def get_token(server, resource, scope):  # pylint: disable=unused-argument
         import adal
         try:
-            return Profile(cli_ctx=cli_ctx).get_raw_token(resource)[0]
+            return Profile(cli_ctx=cli_ctx).get_raw_token(resource=resource,
+                                                          subscription=cli_ctx.data.get('subscription_id'))[0]
```

Comment:
The Key Vault data-plane client factory has been duplicated in several other places which all face the same issue: https://github.com/Azure/azure-cli/pull/17812.

We should extract one common Key Vault client factory, so that 

- the fix applies to all command modules which use Key Vault
- it facilitates the future track 2 migration

Reply:
Done~